### PR TITLE
Refactor introspection client into components for clarity

### PR DESCRIPTION
**Pull request by juliettepretot**
# Checklist

- [ ] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [ ] I have checked that these tests are run by
        [Cloudbuild](/cloudbuild.yaml)
  - [ ] I have updated [documentation](/docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [ ] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

**Review by conradgrobler — APPROVED** [review]

File: oak_runtime/introspection_browser_client/index.tsx [hunk 1]
```diff
@@ -16,236 +16,6 @@
 
 import React from 'react';
 import ReactDOM from 'react-dom';
-import introspectionEventsProto from './proto/introspection_events_pb';
+import Root from '~/components/Root';
```
Comment (tiziano88) [inline oak_runtime/introspection_browser_client/index.tsx:19]: Is it common practice to use `~` to refer to the root of the hierarchy? I have never seen this.

File: oak_runtime/introspection_browser_client/components/Root/index.tsx [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+  const eventType = event.getEventDetailsCase();
+  const { EventDetailsCase } = introspectionEventsProto.Event;
+
+  switch (eventType) {
+    case EventDetailsCase.NODE_CREATED:
+      applicationState.nodeInfos.set(event!.getNodeCreated()!.getNodeId(), {
+        abiHandles: new Map()
+      });
+
+      break;
```
Comment (tiziano88) [inline oak_runtime/introspection_browser_client/components/Root/index.tsx:88]: Are these breaks actually necessary? Does typescript have implicit fall-through?

**Review by tiziano88 — APPROVED** [review]

File: oak_runtime/introspection_browser_client/components/Root/index.tsx (hunk 2 above)
Comment (juliettepretot) [reply oak_runtime/introspection_browser_client/components/Root/index.tsx:88]: Sadly yes

Comment (juliettepretot) [reply oak_runtime/introspection_browser_client/components/Root/index.tsx:88]: image
Cursed JavaScript behaviour

Comment (tiziano88) [reply oak_runtime/introspection_browser_client/components/Root/index.tsx:88]: The stuff of nightmares

File: oak_runtime/introspection_browser_client/index.tsx (hunk 1 above)
Comment (juliettepretot) [reply oak_runtime/introspection_browser_client/index.tsx:19]: With TS/JS being an organically grown ecosystem, there isn't a clear convention for this. `~` is one of the more common options (as used by [Parcel](https://parceljs.org/module_resolution.html). Other options include `@components/Root` or simply `components/Root`. 

I personally prefer `~` as it does not usually conflict with other paths or NPM packages.

File: oak_runtime/introspection_browser_client/components/Root/index.tsx (hunk 2 above)
Comment (juliettepretot) [reply oak_runtime/introspection_browser_client/components/Root/index.tsx:88]: image

**Pull request merged**